Merge entire new transfer system in main (#71)
aurora-multiphysics/hippo: doc/content/step_by_step.md

```toml
# fluid.i

[FoamVariables]
    [fluid_wall_temp]
        type = FoamVariableField
        foam_variable = T
    []
[]
```
Internally, Hippo creates `AuxVariable`s which can use MOOSE's transfer system and so can be transferred between MOOSE apps.

### Imposing boundary conditions on OpenFOAM from MOOSE

Hippo also implements the `FoamBC` system that allows you to impose boundary conditions on OpenFOAM from the Hippo input file. In this case, we wish to impose a heat flux BC on the temperature field. This requires the `FoamFixedGradientBC`, which mirrors OpenFOAM's `fixedGradient` BC type. As in this case, we provide a heat flux, we must also provide the thermal conductivity, which is specified as the `diffusivity_coefficient`. `kappa` is the name of the thermal conductivity variable in OpenFOAM.

```toml
# fluid.i

[FoamBCs]
    [solid_heat_flux]
        type = FoamFixedGradientBC
        foam_variable = T
        diffusivity_coefficient = kappa
    []
[]
```

> Similar to `FoamVariable`, this creates an `AuxVariable` of the same name under the hood, so the transfer system can be used to provide value of the heat flux to imposed on OpenFOAM.

Reviewer: This sentence could do with a re-work:

> _of the same name_
  
The same name as what?

> _provide value of the heat flux to imposed_

Some mixed-up/typo-ed grammar here.

Suggestion below, but feel free to re-phrase as you like:

```suggestion
Similar to `FoamVariable`, this creates an `AuxVariable` with name `solid_heat_flux` under the hood. 
MOOSE's transfers system can be used to set the values in the `AuxVariable`, that are then imposed on the OpenFOAM solver's boundary.
```
